src/PolyExecution/HttpRequestInput.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace SilverStripe\PolyExecution;
+
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Core\Injector\Injectable;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Input that populates options from an HTTPRequest
+ *
+ * Use this for inputs to PolyCommand when called from a web request.
+ */
+class HttpRequestInput extends ArrayInput
+{
+    use Injectable;
+
+    protected bool $interactive = false;
+
+    /**
+     * @param array<InputOption> $commandOptions Any options that apply for the command itself.
+     * Do not include global options (e.g. flush) - they are added explicitly in the constructor.
+     */
+    public function __construct(HTTPRequest $request, array $commandOptions = [])
+    {
+        $definition = new InputDefinition([
+            // Also add global options that are applicable for HTTP requests
+            new InputOption('quiet', null, InputOption::VALUE_NONE, 'Do not output any message'),
+            new InputOption('verbose', null, InputOption::VALUE_OPTIONAL, 'Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug'),
```

Comment:
Does this match how symfony console apps normally do verbosity? Composer does -v / -vv / -vvv - is that more standard? --verbose=2 reads a little weird to me